[quant][graphmode][fx] Skip observering boolean Tensors

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57375 [quant][graphmode][fx] Skip observering boolean Tensors**

Summary:
Skip observing the input for masked_fill. Currently we don't have a way to
query the type of Proxy in GraphModule, hopefully we should have the functionality to annotate the type,
we'll need to annotate a Proxy to be a boolean Tensor to remove this hack.

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_boolean_tensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28126003](https://our.internmc.facebook.com/intern/diff/D28126003)

jerryzh168 committed:
[quant][graphmode][fx] Skip observering boolean Tensors

Summary:
Skip observing the input for masked_fill. Currently we don't have a way to
query the type of Proxy in GraphModule, hopefully we should have the functionality to annotate the type,
we'll need to annotate a Proxy to be a boolean Tensor to remove this hack.

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_boolean_tensor

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Skip observering boolean Tensors"


Summary:
Skip observing the input for masked_fill. Currently we don't have a way to
query the type of Proxy in GraphModule, hopefully we should have the functionality to annotate the type,
we'll need to annotate a Proxy to be a boolean Tensor to remove this hack.

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_boolean_tensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28126003](https://our.internmc.facebook.com/intern/diff/D28126003)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Skip observering boolean Tensors"


Summary:
Skip observing the input for masked_fill. Currently we don't have a way to
query the type of Proxy in GraphModule, hopefully we should have the functionality to annotate the type,
we'll need to annotate a Proxy to be a boolean Tensor to remove this hack.

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_boolean_tensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28126003](https://our.internmc.facebook.com/intern/diff/D28126003)

[ghstack-poisoned]

vkuzo commented on torch/quantization/fx/quantize.py:
```diff
@@ -583,6 +583,21 @@ def load_arg(a: Argument) -> Argument:
+        # it is boolean Tensor
+        # TODO: we need type info from proxy, what type of Tensor this is
+        # similar to TorchScript
+        if (node.op, node.target) == ("call_method", "masked_fill"):
```
could we instead structure this as something like https://github.com/pytorch/pytorch/blob/master/torch/quantization/fx/utils.py#L406, 

```
if second_input_arg_is_boolean_tensor(node):
  ...
```

jerryzh168 replied:
sure

jerryzh168 committed:
Update on "[quant][graphmode][fx] Skip observering boolean Tensors"


Summary:
Skip observing the input for masked_fill. Currently we don't have a way to
query the type of Proxy in GraphModule, hopefully we should have the functionality to annotate the type,
we'll need to annotate a Proxy to be a boolean Tensor to remove this hack.

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_boolean_tensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28126003](https://our.internmc.facebook.com/intern/diff/D28126003)

[ghstack-poisoned]
